File: src/infrastructure/FollowersRepository.js
var _ = require('lodash');

var FollowersRepository = function FollowersRepository(){
    var projections = [];

    this.save = function save(projection){
        var hasFollower = _.some(projections, function(item) {
            return item.followee.equals(projection.followee) && item.follower.equals(projection.follower);
        });

        if(!hasFollower){
            projections.push(projection);
        }
    };

    this.remove = function(projection){
        _.remove(projections, projection);
    };

    this.getFollowers = function getFollowers(userId){
        return _.chain(projections).filter(function(projection){
            return projection.followee.equals(userId);
        }).map(function(projection){
            return projection.follower;
        }).value();
    };
};

exports.create = function create(){
    return new FollowersRepository();
};

File: Mixter.Infrastructure/FollowersRepository.cs
using System.Collections.Generic;
using System.Linq;
using Mixter.Domain.Core.Subscriptions;
using Mixter.Domain.Identity;

namespace Mixter.Infrastructure
{
    public class FollowersRepository
    {
        private readonly IList<FollowerProjection> _projections = new List<FollowerProjection>();

        public void Save(FollowerProjection projection)
        {
            _projections.Add(projection);
        }

        public void Remove(FollowerProjection projection)
        {
            _projections.Remove(projection);
        }

        public IEnumerable<UserId> GetFollowers(UserId followee)
        {
            return _projections.Where(o => o.Followee.Equals(followee)).Select(o => o.Follower);
        }
    }
}
